src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -691,11 +734,34 @@ private void generateBody(Body body) {
                         processOperands(op, isLastOpResultOnStack);
                         cob.checkcast(((JavaType) op.type()).toNominalDescriptor());
                     }
+                    case LambdaOp op -> {
+                        // @@@ double the captured values to enable LambdaMetafactory.FLAG_QUOTABLE
+                        for (Value cv : op.capturedValues()) {
+                            load(cv);
+                        }
```

Comment:
You should only need that when the resolved functional interface class is assignable to `Quotable`. This code appears to be quoting all lambdas?

In your test code you have a functional interface `Func` but it does not extend from `Quotable`, so when generating the bytecode you don't need to apply the quoting logic for lambda expressions targeted to that interface. The test could should be expanded ti also include say `interface FuncQuotable extends FuncQuotable { ... }`, and from such instances their code model can be obtained.

Reply:
Yes, now it is quoting all lambdas.
We may restrict it to interfaces explicitly extending `Quotable`.
Or we may use the ability of `LMF` to inject additional interfaces and make all lambdas generated from code models as `Quotable`.

Reply:
We should only quote when the the FIs extend from `Quotable` or intersects with it. Otherwise, for non-quotable lambdas we are doing unnecessary work, deviating from what the compiler does, and potentially deviating from required behavior.

At the moment we don't support intersection types, but we plan to. I would suggest setting up the code path for quotable and non-quotable and switching based on whether the FI class is assignable. Later on we can enhance when intersection types are supported.